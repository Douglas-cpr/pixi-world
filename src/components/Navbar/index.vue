<script lang="ts" setup>
import Icon from '@/components/factories/Icon/index.vue'
import Button from '@/components/Button/index.vue'
import { useModal } from '@/hooks/useModal'
import store from '@/store'

const modal = useModal()

const openLoginModal = () => {
  modal.open({
    component: 'ModalLogin'
  })
}

const openModalSignup = () => {
  modal.open({
    component: 'ModalSignup'
  })
}
</script>

<template>
  <header class="header">
    <Icon name="Logo"/>
    <ul class="header__routes">
      <li :class="{ 'route__link--selected': true }" class="route__link"><a>Home</a></li>
      <li class="route__link"><a>About</a></li>
      <li class="route__link"><a>Features</a></li>
    </ul>
    <div class="header__actions" v-if="!store.currentUser">
      <Button type="primary" :onClick="openLoginModal" class="actions__btn">Sign In</Button>
      <Button type="secondary" :onClick="openModalSignup" :width="140" class="actions__btn">Create Account</Button>
    </div>
    <div class="header__user" v-else>
      <div class="user__name">
        <span>{{ store.currentUser.username }}</span>
      </div>
    </div>
  </header>
</template>

<style scoped>
.header {
  display: flex;
  justify-content: space-between;
  align-items: center;
  height: 60px;
  background-color: var(--brand-main);
  color: var(--text-main);
}
.header__routes {
  display: flex;
  gap: 1.5rem;
  list-style: none;
  cursor: pointer;
}

.header__actions {
  display: flex;
  gap: 10px;
  cursor: pointer;
}

.actions__btn {
  cursor: pointer;
}

.route__link {
  transition: all ease-out var(--transition-seconds);
}

.route__link:hover {
  color: var(--link-hover);
}

.route__link {
  display: flex;
  justify-content: center;
  align-items: center;
  position: relative;
}

.route__link--selected {
  color: var(--link-selected);
}

.route__link--selected::after {
  content: '';
  width: 10px;
  height: 10px;
  top: 25px;
  position: absolute;
  border-radius: 999px;
  background-color: var(--link-hover);
}

.user__name {
  font-weight: bold;
}

@media (max-width: 425px) {
  .header {
    flex-direction: column;
    gap: 2.5rem;
    height: auto;
  }
}
</style>