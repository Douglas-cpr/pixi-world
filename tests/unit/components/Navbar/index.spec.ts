import Navbar from '@/components/Navbar/index.vue'
import { shallowMount, VueWrapper } from '@vue/test-utils'

describe('<Navbar />', () => {
  let wrapper: VueWrapper<any>

  const createComponent = () => {
    wrapper = shallowMount(Navbar)
  }

  const findLinkList = () => wrapper.find('ul')

  afterEach(() => {
    wrapper.unmount()
  })

  it ('renders correctly', () => {
    createComponent()
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should have a link to home page', () => {
    createComponent()

    const linkList = findLinkList()

    expect(linkList.text()).toContain('Home')
  })

  it('should have a link to about page', () => {
    createComponent()

    const linkList = findLinkList()

    expect(linkList.text()).toContain('About')
  })

  it('should have a link to features page', () => {
    createComponent()

    const linkList = findLinkList()

    expect(linkList.text()).toContain('Features')
  })
})