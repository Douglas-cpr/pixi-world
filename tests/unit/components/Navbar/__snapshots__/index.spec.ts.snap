// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar /> renders correctly 1`] = `
<header
  class="header"
>
  <icon-stub
    name="Logo"
  />
  <ul
    class="header__routes"
  >
    <li
      class="route__link--selected route__link"
    >
      <a>
        Home
      </a>
    </li>
    <li
      class="route__link"
    >
      <a>
        About
      </a>
    </li>
    <li
      class="route__link"
    >
      <a>
        Features
      </a>
    </li>
  </ul>
  <div
    class="header__actions"
  >
    <button-stub
      class="actions__btn"
      disabled="false"
      height="35"
      type="primary"
      width="100"
    />
    <button-stub
      class="actions__btn"
      disabled="false"
      height="35"
      type="secondary"
      width="140"
    />
  </div>
</header>
`;
